components/external_storage/src/s3.rs

```diff
@@ -119,7 +120,14 @@ impl ExternalStorage for S3Storage {
             storage_class: get_var(&self.config.storage_class),
             ..Default::default()
         };
-        futures::executor::block_on(self.client.put_object(req))
+        let mut runtime = Runtime::new().map_err(|e| {
```

Comment:
Can it be shared by other requests? I'm afraid too many threads will be spawned.

Reply:
Is share inside `S3Storage` enough, or shall it be shared at higher level?

Reply:
or we can explicitly control number of threads for the runtime, but I don't know a good number.

Reply:
I think inside `S3Storage` should be enough.

Reply:
Could it be `Future::wait` in the current thread?

Reply:
> Could it be `Future::wait` in the current thread?

seems `Future` doesn't have a `wait` method?

Reply:
I see, it's future 0.3 now.

Reply:
I think this is not resolved. Maybe I should use request for change next time :(.

Reply:
> I think this is not resolved. Maybe I should use request for change next time :(.

I'll try a bit to address it tomorrow.